test/javascript/components/concept-map/ConceptMap.test.tsx

```diff
@@ -1,105 +1,94 @@
 // Deps
 import React from 'react'
+import '@testing-library/jest-dom'
 
 // Test deps
-import {
-  getByTestId,
-  getByText,
-  getByTitle,
-  queryByTitle,
-  render,
-} from '@testing-library/react'
+import { screen, render, waitFor } from '@testing-library/react'
 import '@testing-library/jest-dom/extend-expect'
 
 // Component
 import { ConceptMap } from '../../../../app/javascript/components/concept-map/ConceptMap'
-import {
-  ConceptStatus,
-  IConcept as Concept,
-  ConceptConnection,
-  ConceptLayer,
-} from '../../../../app/javascript/components/concept-map/concept-map-types'
 
 describe('<ConceptMap />', () => {
   test('renders empty component', () => {
-    const { container } = renderMap([], [], [], {})
+    const { container } = render(
+      <ConceptMap
+        concepts={[]}
+        levels={[[]]}
+        connections={[]}
+        status={{}}
+        exercise_counts={{}}
+      />
+    )
     const map = container.querySelector('.c-concepts-map')
     expect(map).not.toBeNull()
   })
 
-  test('renders single incomplete concept', () => {
+  test('renders single incomplete concept', async () => {
     const testConcept = concept('test')
-    const { container } = renderMap([testConcept], [[testConcept.slug]], [], {
-      test: 'locked',
-    })
-    const conceptEl = getByText(container, 'Test')
-    const completeIconEl = queryByTitle(
-      container,
-      'You have mastered this concept'
+
+    render(
+      <ConceptMap
+        concepts={[testConcept]}
+        levels={[[testConcept.slug]]}
+        connections={[]}
+        status={{
+          test: 'unavailable',
+        }}
+        exercise_counts={{
+          test: {
+            exercises: 1,
+            exercises_completed: 0,
+          },
+        }}
+      />
     )
-    expect(completeIconEl).toBeNull()
+
+    await waitFor(() => {
```

Comment:
Does the bottom one (`"You have mastered this concept"`) appear on initial mount but disappear after? Otherwise it shouldn't be in this wait for. You might want to add a comment _why_ this waitFor is necessary for future readers.

Reply:
The title comes from the "completed" svg checkmark, so it shouldn't appear on a concept that isn't complete (the second test) and should appear when it is complete (the third test)

Reply:
So they all appear when the concept is rendered, but doesn't appear visible at first

Reply:
If it's not visible, do you need to wait for it to not be visible?

Reply:
So the only test indicator that a concept is complete or not is the svg checkmark and I wanted to write a test to make sure that incomplete concepts don't have an svg checkmark.  Should I not do that?

Reply:
Okay. I get it now:

1. the `waitFor` you need because it might still be triggering re-renders. Only put inside the `waitFor` what will be the _end_ state. In this case, the thing is now visible.
2. then you want a "invariant" check. It will be a regular assertion that checks that the icon is NOT visible.
3. now you do the interaction
4. now you _wait for_ the interaction to complete
5. if the _wait for_ isn't checking that the icon becomes visible, you should assert here, otherwise you're done.

So basically what I'm saying is: separate the code that waits for the component to _come to rest_ from the code that is testing an invariant. Your idea is solid.